e2e/testenv/fail.go

```diff
@@ -0,0 +1,35 @@
+package testenv
+
+import (
+	"os"
+	"os/exec"
+	"path"
+
+	ginkgo "github.com/onsi/ginkgo/v2"
+)
+
+func FailAndGather(message string, callerSkip ...int) {
+	if _, err := exec.LookPath("crust-gather"); err != nil {
+		ginkgo.GinkgoWriter.Print("⛔ crust-gather is not available, not dumping cluster info")
+		ginkgo.Fail(message, callerSkip...)
+	}
+
+	pwd := os.Getenv("GITHUB_WORKSPACE")
+	if pwd != "" {
+		pwd = path.Join(pwd, "tmp", "upstream")
```

Comment:
Is this why we call `crust-gather` explicitly for downstream clusters through:
```
kubectl config use-context k3d-downstream
crust-gather collect --exclude-namespace=kube-system --exclude-kind=Lease --duration=5s -f tmp/downstream
```
... but not for upstream clusters?
It looks pretty likely from the command run below, but just to be sure.

Reply:
I didn't want to make the fail handler downstream cluster aware. Downstreams are just dumped in the workflows for now.